src/platforms/esp32/components/avm_sys/platform_nifs.c

```diff
@@ -492,6 +492,25 @@ static term nif_esp_get_mac(Context *ctx, int argc, term argv[])
     return term_from_literal_binary(mac, 6, &ctx->heap, ctx->global);
 }
 
+static term nif_esp_get_default_mac(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+    UNUSED(argv);
+
+    uint8_t mac[6];
+    esp_err_t err = esp_efuse_mac_get_default(mac);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "Unable to read default mac.  err=%i", err);
+        RAISE_ERROR(ERROR_ATOM);
```

Comment:
Fine by me. I'll add a TODO in this PR and create an issue about the API change.

Reply:
Sorry for the confusion here, I'd rather go with `{ok, Result} | {error, Reason}` already with this function.
So we reduce API changes in advance (and we make less boring future work for adapting existing functions also for us).

Reply:
Alright, and I'll include the API change for `change_mac` in the current PR too, right?

Reply:
I suggest the following:
- adopt the new API for `esp:get_default_mac/0` for the reason I mentioned. It is a new function so no breaking API change will happen
- keep `get_mac/1` as is, since changing return type would break existing API, and we are approaching v0.6.0 so I'd like to keep existing API frozen.